### ISSUE TITLE
Configure AppVeyor build to include variants

### ISSUE THREAD

**Issue by ddugovic** [ddugovic/Stockfish#445]
It is my understanding that the Travis CI build uses `src/Makefile` however the AppVeyor build does not.

**Comment by ppigazzini** [comment]
Try to ask to @niklasf, he uses a sophisticated Travis CI and AppVeyor configuration to make the releases of [fishnet](https://github.com/niklasf/Stockfish) for Windows, Linux and OSX.

**Comment by ddugovic** [comment]
Thanks @ppigazzini , I do have some guidance but haven't been able to get to this yet.

**Comment by niklasf** [comment]
We now have an AppVeyor build for variants: https://github.com/ddugovic/Stockfish/blob/master/deploy/msys.sh

Right now it only profile-builds binaries. It could be extended with tests. (I am not sure if that's nescessary. I don't think we ever had a Windows-only bug.)

Should we maybe link the downloadable binary snapshots in the README or the Wiki?

**Comment by Nordlandia** [comment]
So all i have to do is paste _msys.sh_ text into mingw64.exe right?

**Comment by ppigazzini** [comment]
@Nordlandia AppVeyor and TravisCI are tools for Continuous Integration (for Windows and Linux platform), check the GitHub project homepage ([click here](https://github.com/ddugovic/Stockfish#overview)), you should view two green icons with the text "build: passing".
They could be used on GitHub for these goals:
1. build a binary for each any single code commit to test the compile process
2. execute the binary and run some tests, to check for errors in execution
3. save the binary under the tab "Release"

If you are interested in compiling yourself multivariant stockfish:

1. write the file C:\msys64\home\YOUR_USERNAME\makefish.sh (change YOUR_USERNAME) with this text:

```bash
#!/bin/bash
# makefish.sh

# update and install packages: after the first run you can comment this instruction
# pacman -S --noconfirm unzip make mingw-w64-x86_64-gcc

# download the Stockfish source code
wget https://github.com/ddugovic/Stockfish/archive/master.zip
unzip master.zip
cd Stockfish-master/src

# build the executable
# keep only the section that fits your CPU architecture, delete the other sections 

# use this section with CPU older than Sandy Bridge:
make profile-build ARCH=x86-64 COMP=mingw
strip stockfish.exe
mv stockfish.exe ../../stockfish_x64.exe
make clean

# use this section with Sandy Bridge and Ivy Bridge CPU:
make profile-build ARCH=x86-64-modern COMP=mingw
strip stockfish.exe
mv stockfish.exe ../../stockfish_x64_modern.exe
make clean

# use this section with Haswell and newer CPU:
make profile-build ARCH=x86-64-bmi2 COMP=mingw
strip stockfish.exe
mv stockfish.exe ../../stockfish_x64_bmi2.exe
make clean

cd
```
2. open a **MSYS2 MinGW 64-bit** console
3. execute this command `pacman -S --noconfirm unzip` to install unzip. You need to run this command only once, don't run it in future
4. execute this command `bash makefish.sh` to build multivariant stockfish

**Comment by ddugovic** [comment]
Yes, linking to downloadable binary snapshots in the README and/or the Wiki sounds useful.

**Comment by Nordlandia** [comment]
Confusing. Please share x64 BMI2 build :)

**Comment by ppigazzini** [comment]
@Nordlandia 
Here the zip with compiles for all CPU: [171117-mv.stockfish_x64_all.zip](https://github.com/ddugovic/Stockfish/files/1482844/171117-mv.stockfish_x64_all.zip)
[VirusTotal check](https://www.virustotal.com/#/file/f77bd07a97c8e3e88e38979e6bcd268b816dfdd3c5555fc1773ff1fd27f20433/detection)

**Comment by ppigazzini** [comment]
@Nordlandia 
I moved my posts in a new Issue (click here https://github.com/ddugovic/Stockfish/issues/457) to keep clean from OT this Issue

**Comment by ppigazzini** [comment]
@ddugovic @niklasf added the links also in the [wiki page](https://github.com/ddugovic/Stockfish/wiki#welcome-to-the-multi-variant-stockfish-wiki)
